monkey/infection_monkey/exploit/tools/brute_force_exploiter.py

```diff
@@ -0,0 +1,101 @@
+import logging
+from io import BytesIO
+from typing import Callable
+
+from common.types import Event
+from infection_monkey.exploit import IAgentBinaryRepository
+from infection_monkey.i_puppet import ExploiterResultData, TargetHost
+from infection_monkey.propagation_credentials_repository import IPropagationCredentialsRepository
+from infection_monkey.utils.threading import interruptible_iter
+
+from . import (
+    IRemoteAccessClient,
+    IRemoteAccessClientBuilder,
+    RemoteAuthenticationError,
+    RemoteFileCopyError,
+    generate_brute_force_credentials,
+)
+from .helpers import get_agent_dst_path
+
+logger = logging.getLogger(__name__)
+
+
+class BruteForceExploiter:
+    def __init__(
+        self,
+        build_command: Callable[[str, str], str],
+        exploit_client_builder: IRemoteAccessClientBuilder,
+        propagation_credentials_repository: IPropagationCredentialsRepository,
```

Comment:
This should just accept a callable that returns all credentials to try. We can provide a callable class in `tools/` that queries the repository and calls `generate_brute_force_credentials()` and construct an instance of it in `plugin.py`.